google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/reframing/ReframingResponseObserver.java

```diff
@@ -209,12 +217,9 @@ protected void onResponseImpl(InnerT response) {
    */
   @Override
   protected void onErrorImpl(Throwable t) {
-    synchronized (lock) {
-      if (error == null) {
-        error = t;
-      }
-      closeOnDone = true;
-    }
+    // order of assignment matters
```

Comment:
I don't completely understand why this isn't synchronized anymore.

Reply:
Only the grpc thread can set this value and it always does this before writing the volatile `done` which ensure cache coherency for whatever thread might be in the delivery loop.

Previously `cancel` would share the `error` variable which meant that there were 2 writers: grpc + an application thread explicitly setting it via cancel. This is longer the case as the cancellation uses its own Atomic

Reply:
Got it!